commons/src/main/java/org/infinispan/commons/dataconversion/BinaryEncoder.java

```diff
@@ -53,7 +53,7 @@ private static boolean isTypeExcluded(Class<?> type) {
    }
 
    private boolean skipEncoding(Object source) {
-      return isTypeExcluded(source.getClass());
+      return source == null || isTypeExcluded(source.getClass());
```

Comment:
In this case all `Encoder` implementation must be able to support null. This means that `UTF8Encoder` is broken without that.